spec.bs

```diff
@@ -1502,7 +1524,70 @@ The <a spec="turtledove">validate and convert auction ad config</a> steps are
 modified to add the following steps just before the last step ("Return
 <var ignore>auctionConfig</var>"), renumbering the later step as appropriate:
 <div algorithm="protected-audience-validate-config-monkey-patch">
-7. If |config|[{{AuctionAdConfig/privateAggregationConfig}}] [=map/exists=]:
+31. If |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"]
+    [=map/exists=]:
+    1. Let |interestGroupBuyers| be |auctionConfig|'s <a spec="turtledove"
+        for="auction config">interest group buyers</a>.
+    1. If |interestGroupBuyers| is null, set |interestGroupBuyers| to a new
+        [=list=].
+    1. [=list/For each=] |index| of [=the exclusive range|the range=] 0 to
+        |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"]'s
+        [=list/size=], exclusive:
+        1. Let |key| be
+            |config|["<code>{{AuctionAdConfig/auctionReportBuyerKeys}}</code>"][|index|].
+        1. If |key|["{{PAHistogramContribution/bucket}}"] is not in the range
+            [0, 2<sup>128</sup>−1], [=exception/throw=] a {{TypeError}}.
+        1. If |index| is equal to or greater than |interestGroupBuyers|' [=list/
```

Comment:
[the implementation](https://crsrc.org/c/third_party/blink/renderer/modules/ad_auction/navigator_auction.cc;drc=738f1250512a5e6ff8cacb6b36540cd50e8156d8;l=2067) seems not doing this, but just simply copying all keys.
So there's a difference in implementation that the step above will continue throw errors if the condition meets for extra |index|.

Reply:
Switched this to a continue and added a note to match the validation logic. And yeah I thought it might be easier to follow in the spec if we didn't just keep these as a list. We could consider switching to more closely match the implementation structure.